src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -169,11 +169,11 @@ private void AddDocumentationContent(ISymbol symbol)
 
                 _documentationMap.Add(
                     SymbolDescriptionGroups.Documentation,
-                    symbol.GetDocumentationParts(_semanticModel, _position, formatter, CancellationToken).ToImmutableArray());
+                    symbol.GetDocumentationParts(_semanticModel, _position, formatter, CancellationToken));
 
                 _documentationMap.Add(
                     SymbolDescriptionGroups.RemarksDocumentation,
-                    symbol.GetRemarksDocumentationParts(_semanticModel, _position, formatter, CancellationToken).ToImmutableArray());
+                    symbol.GetRemarksDocumentationParts(_semanticModel, _position, formatter, CancellationToken));
```

Comment:
This only avoids creating a new array since `GetDocumentationParts` already returns an immutable array.

Reply:
`ToImmutableArray()` won't clone the array if the input is already an immutable array, but this does avoid boxing the `IEnumerable<T>` for the call and is cleaner so 👍

Reply:
> ToImmutableArray() won't clone the array if the input is already an immutable array,

@sharwell Yeah that seem the case:

https://github.com/dotnet/runtime/blob/59d9d0d16858635a8c228be2ab7ad8882b0d1c71/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs#L372-L380

So the docs of CA2009 are inaccurate/incorrect?

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2009

> Performance issue: Unnecessary creation of a duplicate immutable collection. The original collection was already immutable and can be used directly.